### PR TITLE
Add Husky pre-commit and pre-push hooks

### PR DESCRIPTION
**Pull request by radiocontrolled**
Resolves https://github.com/BBC-News/psammead/issues/24

Adds Husky hooks to the package.json that will prevent `git commit` if there are linting errors, and `git push` if there are testing errors.

```
  "husky": {
    "hooks": {
      "pre-commit": "npm run test:lint && npm run test:lint:css",
      "pre-push": "npm run test:unit"
    }
  }
```

**Developer-suggested testing ideas**
`pre-commit` test: 
- Make an error that will cause a linting issue, for example changing this ExampleButton from `const` to `var`, [here](https://github.com/BBC-News/psammead/blob/latest/packages/example-button/src/index.jsx#L5)
- try to `git commit` that change to github - you should be preventing from making the commit.
- undo that change

`pre-push` test:
- Make an error that will cause a test failure, such as changing [this line to equal 2](https://github.com/BBC-News/psammead/blob/latest/packages/example-button/src/index.test.jsx#L13)
- `git add` that error
- `git commit` the breaking change
- Attempt to `git push` - the push should be rejected due to the failing test and you should see the error `error: failed to push some refs to 'git@github.com:BBC-News/psammead.git'`

- [ ] Tests added for new features
- [ ] Test engineer approval

### CODE REVIEW

File: .eslintignore [hunk 1]
```diff
@@ -1 +1,2 @@
 packages/*/node_modules
+packages/*/dist
```
Comment (dr3) [inline .eslintignore]:
```suggestion
packages/*/dist

```

Comment (radiocontrolled) [reply .eslintignore]: @dr3 fixed that 👍

**Review by dr3 — APPROVED** [review]

**Review by sareh — APPROVED** [review]
👍 Looks great.

**Review by bcmn — APPROVED** [review]

**Review by pjlee11 — APPROVED** [review]

**Comment by jamesbrumpton** [comment]
LGTM

**Pull request merged**